### DISCUSSION TITLE
Conditional block rendering

### DISCUSSION THREAD

**Discussion by tadious** [quillforms/quillforms#92] [Q&A] [answered]
Is it possible to render/skip blocks based on responses from previous blocks?

**Comment by mdmag** [comment, marked as answer]
Yes, if you are using React, you have a full control on passed blocks.
If you are using WordPress, this feature is on our pro version.

**Comment by tadious** [comment]
Yes I meant using React. I couldn't find any documentation for that. Conditional as in, show a question based on responses from previous answers?

**Comment by AgrPaavan** [comment]
Hey @tadious. Have you found any solutions for conditional block rendering?

**Reply by tadious** [reply]
@AgrPaavan No, I didn't, but the answer below will achieve it. 💯

**Comment by mdmag** [comment]
@tadious @AgrPaavan, 
You pass the blocks as a prop to the `Form` Component. 
You can have a state for these blocks and using these useful hooks, you can achieve conditional block rendering:

1- `useAnswers`
`import { useAnswers } from "@quillforms/renderer-core";`
`const formAnswers = useAnswers();`

2- `useFieldAnswer`
`import { useFieldAnswer } from "@quillforms/renderer-core";`
`const fieldAnswer = useFieldAnswer( FieldId );`

**Reply by tadious** [reply]
@mdmag Great thanks

**Reply by ahmad-punch** [reply]
Could you please add an example like this one on code sandbox? https://codesandbox.io/s/quill-forms-custom-block-registration-xmbyiy

Also, I took a look inside quillforms/renderer-core with vscode and searched useAnswers but was not able to find anything inside :(

**Reply by mdmag** [reply]
@ahmad-punch 
I have added a doc for that:
https://github.com/quillforms/quillforms/blob/master/react-docs/conditional-blocks-rendering.md

Also, I added a codesandbox example:
https://codesandbox.io/s/quill-forms-conditional-blocks-rendering-0r9x2e

Please let me know if it is clear now.

**Comment by maxcook** [comment]
How to do this on Wordpress? @mdmag
Which paid plan is needed, Baisc, Plus or Enterprise?
